eng/validation/test-publishing.ps1

```diff
@@ -33,6 +35,28 @@ function Find-BuildInTargetChannel(
     return $false
 }
 
+$global:arcadeSdkPackageName = 'Microsoft.DotNet.Arcade.Sdk'
+$global:arcadeSdkVersion = $GlobalJson.'msbuild-sdks'.$global:arcadeSdkPackageName
+$global:githubRepoName = "arcade"
+$sha = Get-LatestBuildSha
+$global:targetBranch = "val/" + $global:githubUser + "/arcade-" + $global:arcadeSdkVersion
+
+## Create a branch from the repo with the given SHA.
+Git-Command $global:githubRepoName checkout -b $global:targetBranch $sha
+
+## Get the BAR Build ID for the version of Arcade we want to use in update-dependecies
+$asset = & $darc get-asset --name $global:arcadeSdkPackageName --version $global:arcadeSdkVersion --github-pat $global:githubPAT --azdev-pat $global:azdoToken --password $global:bartoken
+$barBuildIdString = $asset | Select-String -Pattern 'BAR Build Id:'
+$barBuildId = ([regex]"\d+").Match($barBuildIdString).Value
+
+## Make the changes to that branch to update Arcade - use darc
+Set-Location $(Get-Repo-Location $global:githubRepoName)
+& $darc update-dependencies --id $barBuildId --github-pat $global:githubPAT --azdev-pat $global:azdoToken --password $global:bartoken
+
+Git-Command $global:githubRepoName commit -am "Arcade branch - version ${global:arcadeSdkVersion}"
```

Comment:
This looks like code that is already in build-arcadewithrepo LN174-181. I think it would be good if we extracted this to a function that does the asset lookup, updates dependencies, and then commits the result.